cupy/_core/_kernel.pyx

```diff
@@ -45,6 +45,12 @@ cdef inline bint _contains_zero(const shape_t& v) except? -1:
     return False
 
 
+@_util.memoize(for_each_device=True)
+def _get_warpsize():
```

Comment:
Should we provide this function to users as a public API?

Reply:
I think not needed as users can query this via existing API.

Btw, how about using `cupy.cuda.device._get_device().attributes["warpSize"]`?

Reply:
`Device.attributes["WarpSize"]` does not work in the CI environment, so I will revert the implementation to use `runtime.getDeviceProperties`.

```py
>>> from cupy.cuda import device
>>> device._get_device().attributes['WarpSize']
0
```